packages/kbn-config-schema/src/types/duration_type.ts

```diff
@@ -36,16 +38,32 @@ export class DurationType extends Type<Duration> {
       defaultValue = options.defaultValue;
     }
 
-    super(internals.duration(), { ...options, defaultValue });
+    let schema = internals.duration();
+    if (options.min) {
+      schema = schema.min(options.min);
+    }
+    if (options.max) {
+      schema = schema.max(options.max);
+    }
+
```

Comment:
I see this function is performing some validation on the `defaultValue` (convert type and even wrap func call).
Should we do the same with `min` and `max`? I'm totally new to types. What are the `schema.min()` and `schema.max()` doing?

I wonder if we should also enforce `min <= defaultValue <= max`.

Reply:
> Should we do the same with min and max?

it's done, but not at the same place, because of the strange way `joi` works...
But we don't need to wrap function calls are we're not supporting siblings/references for options

> I wonder if we should also enforce min <= defaultValue <= max

We're not doing it for other types having those options (e.g `bytes` and `number`), so I assumed it wasn't necessary and more consistent.